plaso/engine/artifacts_trie.py

```diff
@@ -0,0 +1,299 @@
+                atrifact_path_segments)
+    ]
+
+  def _GetNonEmptyPathSegments(self, path, separator):
+    """Splits a path into segments and remove non-empty segments.
+
+    Args:
+        path (str): The path string to be split.
+        separator (str): The path separator.
+
+    Returns:
+        list[str]: A list of non-empty path segments.
+    """
+    return [s for s in path.split(separator) if s]
+
+  def _GetArtifactsPaths(self, node):
+    """Retrieves a mapping of artifact names to their paths.
+
+    Args:
+      node (TrieNode): current trie node being traversed.
+
+    Returns:
+        dict: dictionary mapping artifact names to their paths.
+    """
+    artifacts_paths = {}
+
+    def _collect_paths(node, current_path, artifacts):
+      """Collects paths from the trie.
+
+      Args:
+          node (TrieNode): current node.
+          current_path (str): path leading to this node.
+          artifacts (dict): dictionary to store artifact paths.
+      """
+      if node.artifacts_names:
+        for artifact_name in node.artifacts_names:
+          path_list = artifacts.setdefault(artifact_name, [])
+          path_list.append(current_path)
+
+      for segment, child_node in node.children.items():
+        # Ensure the path_separator attribute exists.
+        if not hasattr(child_node, 'path_separator'):
+          child_node.path_separator = node.path_separator
+
+        # Construct the next path segment.
+        if current_path == child_node.path_separator:
+          # Means it is the root folder, i.e. `/`
+          next_path = current_path + segment
+        else:
+          next_path = current_path + child_node.path_separator + segment
+
+        _collect_paths(child_node, next_path, artifacts)
+
+    _collect_paths(node, '', artifacts_paths)
+    return artifacts_paths
+
+  def _CustomPathJoin(self, separator, current_path, new_segment):
+    """Joins path components using a custom separator, replacing os.sep.
+
+    Args:
+      separator (str): The custom separator to use.
+      current_path (str): The current path.
+      new_segment (str): The new segment to add to it.
+
+    Returns:
+      str: The joined path with the custom separator.
+    """
+    current_path = current_path.replace(separator, os.sep)
+    joined_path = os.path.join(current_path, new_segment)
+    return joined_path.replace(os.sep, separator)
+
+  def _MatchesGlobPattern(self, glob_pattern, path, path_separator):
+    """Checks if a path matches a given glob pattern.
+
+    Args:
+      glob_pattern: The glob pattern to match against.
+      path: The path to check.
+      path_separator: The path separator used in the glob pattern.
+
+    Returns:
+      True if the path matches the glob pattern, False otherwise.
+    """
+    # Normalize paths using the appropriate separators
+    glob_pattern = glob_pattern.strip(path_separator).split(path_separator)
+    path = path.strip(path_separator).split(path_separator)
+
+    i = 0
```

Comment:
per style guide please don't use variable names that don't relay their purpose, this is done in other places as well